test/jdk/java/lang/Thread/virtual/MonitorWaitNotify.java

```diff
@@ -640,6 +640,60 @@ void testParkingPermitNotOffered() throws Exception {
         assertTrue(completed.get());
     }
 
+    /**
+     * Test no deadlock happens when Object.wait is called from a mix of pinned and non-pinned
+     * paths and notification is done using notifyAll.
+     */
+    @Test
+    void testMixedPinnedUnmounted() throws Exception {
```

Comment:
What would you think of testing timed-wait too? Some of the other tests are paramerized with a value source and some time values to test both untimed and timed waits.

Reply:
Done.